[Outlook] (manifest) Define default SendMode option and add Block option (#3297)
OfficeDev/office-js-docs-pr: docs/reference/manifest/launchevent.md

|:-----|:-----|:-----|
|  **Type**  |  Yes  | Specifies a supported event type. For the set of supported types, see [Configure your Outlook add-in for event-based activation](../../outlook/autolaunch.md#supported-events). |
|  **FunctionName**  |  Yes  | Specifies the name of the JavaScript function to handle the event specified in the `Type` attribute. |

> |  **SendMode** (preview) |  No  | Used by `OnMessageSend` and `OnAppointmentSend` events. Specifies the options available to the user if your add-in stops an item from being sent. If a `SendMode` option is not specified, the `SoftBlock` option is set by default. For available options, refer to [Available SendMode options](#available-sendmode-options-preview). |

Reviewer: ```suggestion
|  **SendMode** (preview) |  No  | Used by `OnMessageSend` and `OnAppointmentSend` events. Specifies the options available to the user if your add-in stops an item from being sent. If the **SendMode** property isn't included, the `SoftBlock` option is set by default. For available options, refer to [Available SendMode options](#available-sendmode-options-preview). |
```

> |  **SendMode** (preview) |  No  | Used by `OnMessageSend` and `OnAppointmentSend` events. Specifies the options available to the user if your add-in stops an item from being sent or if the add-in is unable to connect to the server. If the **SendMode** property isn't included, the `SoftBlock` option is set by default. For available options, refer to [Available SendMode options](#available-sendmode-options-preview). |

Reviewer: ```suggestion
|  **SendMode** (preview) |  No  | Used by `OnMessageSend` and `OnAppointmentSend` events. Specifies the options available to the user if your add-in stops an item from being sent or if the add-in is unavailable. If the **SendMode** property isn't included, the `SoftBlock` option is set by default. For available options, refer to [Available SendMode options](#available-sendmode-options-preview). |
```

## Available SendMode options (preview)

> When you include the `OnMessageSend` or `OnAppointmentSend` event in the manifest, you should also set the **SendMode** property. If a **SendMode** option is not specified, `SoftBlock` is set by default. The following are the available options. Based on the conditions your add-in is looking for, the user is alerted if your add-in finds an issue in the item being sent.

Reviewer: ```suggestion
When you include the `OnMessageSend` or `OnAppointmentSend` event in the manifest, you should also set the **SendMode** property. If the **SendMode** property isn't included, the `SoftBlock` option is set by default. The following are the available options. Based on the conditions your add-in is looking for, the user is alerted if your add-in finds an issue in the item being sent.
```

| SendMode option | Description |
|---|---|
|`PromptUser`|In the alert, the user can choose to **Send Anyway**, or address the issue then try to send the item again.|

> |`SoftBlock`|The user is alerted that the item they're sending doesn't meet the add-in conditions and the add-in will decide what to do. For example, the add-in can be set to simply alert the user about the issue or require the user to address the issue before sending the item again. If the add-in is unable to connect to the server when the add-in is activated, the item will always be sent.|

Reviewer: ```suggestion
|`SoftBlock`|The add-in decides what to do if the item being sent doesn't meet the add-in's conditions. The add-in can be set to simply alert the user about the issue then the user decides if to **Send Anyway** or address the issue. Otherwise, the add-in can be set to require the user to address the issue before they try to send the item again.|
```

> |`SoftBlock`|Default option if the **SendMode** property isn't included. The user is alerted that the item they're sending doesn't meet the add-in conditions, and must address the issue before trying to send the item again. However, if the add-in is unavailable when an item is being sent (for example, the add-in is unable to connect to the server), the user will be able to send the item.

Reviewer: ```suggestion
|`SoftBlock`|Default option if the **SendMode** property isn't included. The user is alerted that the item they're sending doesn't meet the add-in's conditions and they must address the issue before trying to send the item again. However, if the add-in is unavailable (for example, there's an error loading the add-in), the item will be sent.|
```
